src/components/AccountDetails/index.tsx

```diff
@@ -241,7 +241,8 @@ export default function AccountDetails({
     const name = Object.keys(SUPPORTED_WALLETS)
       .filter(
         (k) =>
-          SUPPORTED_WALLETS[k].connector === connector && (connector !== injected || isMetaMask === (k === 'METAMASK'))
```

Comment:
generally agree. was trying to avoid touching this code too much since it's a bit of a rabbithole... the SUPPORTED_WALLETS constant is what's powering this file and WalletModal, but it could use a bit of a re-work. i'd like to move to using `connectionType` as the source of truth like you said, but not in this PR.